File: credit.c
#include <stdio.h>
#include <cs50.h>
bool check_validity(long long credit_card_number);
bool checksum(long long ccn);
int find_length(long long n);
void print_credit_card_brand(long long ccn);

int main(void)
{
    long long credit_card_number;
    do
    {
        credit_card_number = get_long_long("Number: ");
    }
    while (credit_card_number < 0);

    if (check_validity(credit_card_number) == true)
    {
        print_credit_card_brand(credit_card_number);
    }
    else
    {
        printf("INVALID\n");
    }

}

// Evaluating the credit card number
bool check_validity(long long credit_card_number)
{
    int len = find_length(credit_card_number);
    return (len == 13 || len == 15 || len == 16) && checksum (credit_card_number);
}

// Finding the length
int find_length(long long n)
{
    int len;
    for (len = 0; n != 0; n /= 10, len++);
    return len;
}

// Checksum using Luhn's Algorithm
bool checksum(long long ccn)
{
    int sum = 0;
    for (int i = 0; ccn != 0; i++, ccn /= 10)
    {
        if (i % 2 == 0)
        {
            sum += ccn % 10;
        }
        else
        {
            int digit = 2 * (ccn % 10);
            sum += digit / 10 + digit % 10;
        }
    }
    return (sum % 10) == 0;
}

// Printing the credit card brands
void print_credit_card_brand(long long ccn)
{
    // AMEX
    if ((ccn >= 34e13 && ccn < 35e13) || (ccn >= 37e13 && ccn < 38e13))
    {
        printf("AMEX\n");
    }
    // MASTERCARD
    else if (ccn >= 51e14 && ccn < 56e14)
    {
        printf("MASTERCARD\n");
    }
    // VISA
    else if ((ccn >= 4e12 && ccn < 5e12) || (ccn >= 4e15 && ccn < 5e15))
    {
        printf("VISA\n");
    }
    // IF NONE OF THOSE
    else
    {
        printf("INVALID\n");
    }
}

File: credit.py
# Program that checks for valid VISA, American Express, and Mastercard card numbers using Luhn’s Algorithm
import cs50


def main():
    while True:
        credit_card = cs50.get_int("Number: ")
        if (credit_card >= 0):
            break
    # Check if credit card is valid
    check_validity(credit_card)


# Evaluating the credit card number
def check_validity(ccn):
    length = get_length(ccn)
    # Check the length
    # Check_sum
    if (checklength(length) == True and checksum(ccn, length) == True) == False:
        print("INVALID")
        return
    # Check/Print the brand
    print_credit_card_brand(ccn)


# Finding the length
def get_length(n):
    length = 0
    while n > 0:
        length = length + 1
        n = n // 10
    return length


# Checking the length
def checklength(length):
    if (length == 13 or length == 15 or length == 16):
        return True
    return False


# Checksum using Luhn's Algorithm
def checksum(ccn, length):
    summation = 0
    for i in range(length):
        if (i % 2 == 0):
            summation = summation + (ccn % 10)
        else:
            digit = 2 * (ccn % 10)
            summation = summation + (digit // 10 + digit % 10)
        ccn = ccn // 10
    if (summation % 10 == 0):
        return True
    return False


# Printing the credit card brands
def print_credit_card_brand(ccn):

    # AMEX
    if ((ccn >= 34e13 and ccn < 35e13) or (ccn >= 37e13 and ccn < 38e13)):

        print("AMEX")

    # MASTERCARD
    elif (ccn >= 51e14 and ccn < 56e14):

        print("MASTERCARD")

    # VISA
    elif ((ccn >= 4e12 and ccn < 5e12) or (ccn >= 4e15 and ccn < 5e15)):

        print("VISA")

    # IF NONE OF THOSE
    else:

        print("INVALID")


main()